src/langchain_google_alloydb_pg/async_vectorstore.py

```diff
@@ -552,9 +552,11 @@ async def __query_collection(
             query_embedding = f"'{embedding}'"
         stmt = f'SELECT *, {search_function}({self.embedding_column}, {query_embedding}) as distance FROM "{self.schema_name}"."{self.table_name}" {filter} ORDER BY {self.embedding_column} {operator} {query_embedding} LIMIT {k};'
         if self.index_query_options:
-            query_options_stmt = f"SET LOCAL {self.index_query_options.to_string()};"
             async with self.engine.connect() as conn:
-                await conn.execute(text(query_options_stmt))
+                # Set each query option individually
+                for query_option in self.index_query_options.to_string().split(","):
```

Comment:
As discussed offline, this solution works. There is a slight performance issue where we concatenate the string then split the string. To reduce confusion with long term maintenance, I prefer a solution where the QueryOptions class has a new method like "to_parameter(self) -> List[str]", then we can set each parameter in the list